addOns/retest/src/main/java/org/zaproxy/addon/retest/EditAlertDialog.java

```diff
@@ -0,0 +1,170 @@
+                new java.awt.event.WindowAdapter() {
+                    @Override
+                    public void windowOpened(java.awt.event.WindowEvent e) {}
+
+                    @Override
+                    public void windowClosing(java.awt.event.WindowEvent e) {
+                        btnCancel.doClick();
+                    }
+                });
+        pack();
+    }
+
+    /**
+     * This method initializes jPanel
+     *
+     * @return javax.swing.JPanel
+     */
+    private JPanel getJPanel() {
+        if (jPanel == null) {
+            GridBagConstraints gridBagConstraints15 = new GridBagConstraints();
+            java.awt.GridBagConstraints gridBagConstraints13 = new GridBagConstraints();
+
+            javax.swing.JLabel jLabel2 = new JLabel();
+
+            java.awt.GridBagConstraints gridBagConstraints3 = new GridBagConstraints();
+
+            java.awt.GridBagConstraints gridBagConstraints2 = new GridBagConstraints();
+
+            jPanel = new JPanel();
+            jPanel.setLayout(new GridBagLayout());
+            gridBagConstraints2.gridx = 1;
+            gridBagConstraints2.gridy = 5;
+            gridBagConstraints2.insets = new java.awt.Insets(2, 2, 2, 2);
+            gridBagConstraints2.anchor = java.awt.GridBagConstraints.EAST;
+            gridBagConstraints3.gridx = 2;
+            gridBagConstraints3.gridy = 5;
+            gridBagConstraints3.insets = new java.awt.Insets(2, 2, 2, 10);
+            gridBagConstraints3.anchor = java.awt.GridBagConstraints.EAST;
+
+            gridBagConstraints13.gridx = 0;
+            gridBagConstraints13.gridy = 5;
+            gridBagConstraints13.fill = java.awt.GridBagConstraints.HORIZONTAL;
+            gridBagConstraints13.weightx = 1.0D;
+            gridBagConstraints13.insets = new java.awt.Insets(2, 10, 2, 5);
+
+            gridBagConstraints15.weightx = 1.0D;
+            gridBagConstraints15.weighty = 1.0D;
+            gridBagConstraints15.fill = java.awt.GridBagConstraints.BOTH;
+            gridBagConstraints15.insets = new java.awt.Insets(2, 2, 2, 2);
+            gridBagConstraints15.gridwidth = 3;
+            gridBagConstraints15.gridx = 0;
+            gridBagConstraints15.gridy = 2;
+            gridBagConstraints15.anchor = java.awt.GridBagConstraints.NORTHWEST;
+            gridBagConstraints15.ipadx = 0;
+            gridBagConstraints15.ipady = 10;
+
+            jPanel.add(getAlertEditPanel(), gridBagConstraints15);
+            jPanel.add(jLabel2, gridBagConstraints13);
+            jPanel.add(getBtnCancel(), gridBagConstraints2);
+            jPanel.add(getBtnOk(), gridBagConstraints3);
+        }
+        return jPanel;
+    }
+    /**
+     * This method initializes btnStart
+     *
+     * @return javax.swing.JButton
+     */
+    private JButton getBtnOk() {
+        if (btnOk == null) {
+            btnOk = new JButton();
+            btnOk.setText(Constant.messages.getString("alert.add.button.save"));
```

Comment:
It would be better if the add-on used its own messages, this makes core changes harder (like the reports case where messages needed to be added back because an add-on is using them).